server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -1718,6 +1718,9 @@ private static IndexMetadata.Builder restoreOverClosedIndex(IndexMetadata snapsh
             .state(IndexMetadata.State.OPEN)
             .version(Math.max(snapshotIndexMetadata.getVersion(), 1 + currentIndexMetadata.getVersion()))
             .mappingVersion(Math.max(snapshotIndexMetadata.getMappingVersion(), 1 + currentIndexMetadata.getMappingVersion()))
+            .mappingsUpdatedVersion(
+                IndexVersion.max(snapshotIndexMetadata.getMappingsUpdatedVersion(), currentIndexMetadata.getMappingsUpdatedVersion())
```

Comment:
If we are restoring a snapshot, why wouldn't the snapshot version replace the updated version for the index?

Reply:
I was copying this from the code above TBH, is there a situation we would be restoring a snapshot from a later version?

Reply:
No, but the latest mappings version might go backwards? If it stays on the current, it might be later than actually necessary (and thereby limit which nodes the snapshot can be allocated to)

Reply:
Hmm, yes, if its being overridden, then it should be the recorded version in the snapshot